pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -265,11 +265,9 @@ func (d *namespacedResourcesDeleter) updateNamespaceStatusFunc(namespace *v1.Nam
 	if namespace.DeletionTimestamp.IsZero() || namespace.Status.Phase == v1.NamespaceTerminating {
 		return namespace, nil
 	}
-	newNamespace := v1.Namespace{}
-	newNamespace.ObjectMeta = namespace.ObjectMeta
-	newNamespace.Status = *namespace.Status.DeepCopy()
+	newNamespace := namespace.DeepCopy()
```

Comment:
What got lost when we only updated the status? Do you know why the prior authors thought it could be omitted? I noticed that `finalizeNamespace`, below, follows the same pattern but this time only for the spec, not the status.

I may come back and add some comments in my own PR.

Reply:
spec fields which are ignored by the server in real requests but which got stomped when submitting via the fake client in the unit test I added